### PR TITLE
Revert ACME to optional dependency, introduce fallback parallel processing

### LINKED ISSUE #363: Solve conda version
For reasons not entirely clear, right now we require since  v2022.08 a rather cumbersome installation path via explicitly creating a conda env with `python=3.8`. Updating our dependencies to the latest (now hard dependency) acme should resolve this.

### PR DESCRIPTION
**Pull request by tensionhead**
## Changes
- reverted to acme being an optional dependency 
- dask.distributed is now a hard dependency to allow for Syncopy standalone parallel processing
- updating Python version in `pyproject.toml`
- removed `fulltests` option, in rare cases the fulltests became the default, in most cases they were just kicked out
- removed remnants of long deprecated functionality from decorators: no support for multiple Syncopy data objects as frontend input
## How it works
- Syncopy now picks up any running Dask client on its own to allow for independent parallel processing via `client.map()`
- with `parallel=True` and no ACME around, a LocalCluster gets spawned as a fallback
- the entire testing pipeline including parallel tests now runs in under 10min
- latest ACME dev is compatible with the updated `syncopy.yml` :rocket:

closes #363 and #371 

Author Guidelines
-----------------
- [ ] Is the change set **< 600 lines**?
- [x] Was the code checked for memory leaks/performance bottlenecks?
- [x] Is the code running locally **and** on the ESI cluster?
- [x] Is the code running on all supported platforms?

Reviewer Checklist
------------------
- [x] Are testing routines present?
- [x] Do **parallel** loops have a set length and correct termination conditions?
- [x] Do objects in the global package namespace perform proper parsing of their input? 
- [x] Do code-blocks provide novel functionality, i.e., no re-factoring using builtin/external packages possible?
- [x] Code layout
  - [x] Is the code PEP8 compliant?
  - [x] Does the code adhere to agreed-upon naming conventions?
  - [x] Are keywords clearly named and easy to understand?
  - [x] No commented-out code?
- [x] Are all docstrings complete and accurate?
- [x] Is the CHANGELOG.md up to date?

### CODE REVIEW

File: syncopy/specest/freqanalysis.py [hunk 1]
```diff
@@ -374,7 +374,7 @@ def freqanalysis(data, method='mtmfft', output='pow',
     # --- Padding ---
 
     # Sliding window FFT does not support "fancy" padding
-    if method == "mtmconvol" and isinstance(pad, str):
+    if method == "mtmconvol" and isinstance(pad, str) and pad != defaults['pad']:
```
Comment (dfsp-spirit) [inline syncopy/specest/freqanalysis.py:377]: nice catch

**Comment by dfsp-spirit** [comment]
oh wow, that's a lot of changes. I'll finish it tomorrow.

**Review by dfsp-spirit — APPROVED** [review]
Great, this looks good. I have adapted some minor things, as discussed.

File: pyproject.toml [hunk 2]
```diff
@@ -21,17 +21,17 @@ description = "A toolkit for user-friendly large-scale electrophysiology data an
 authors = ["Stefan Fürtinger <sfuerti@esi-frankfurt.de>", "Tim Schäfer <tim.schaefer@esi-frankfurt.de>", "Joscha Schmiedt <schmiedt@uni-bremen.de>", "Gregor Mönke <gregor.moenke@esi-frankfurt.de>"]
 
 [tool.poetry.dependencies]
-# acme needs hard python version pinning
-python = "^3.8, <3.9"
+python = "^3.8"
```
[the remark this answers is not in the document]
Comment (tensionhead) [reply pyproject.toml:24]: Yes, let's see how the conda deployment then goes.. locally I have a Python 3.10 poetry env now

File: syncopy/tests/test_decorators.py [hunk 3]
```diff
@@ -156,82 +158,93 @@ def test_invalidcallstyles(self):
...
+            groupList = [group_objects(data, groupbychan=letter) for data in self.dataObjs]
+            groupList = [el for el in groupList if el is not None]
             assert len(groupList) == nOccurences
 
             # 1. data positional, 2. cfg positional
             cfg = StructDict()
             cfg.groupbychan = letter
-            fnameList = group_objects(*self.dataObjs, cfg)
+            fnameList = [group_objects(data, cfg) for data in self.dataObjs]
+            fnameList = [el for el in groupList if el is not None]
```
[the remark this answers is not in the document]
Comment (tensionhead) [reply syncopy/tests/test_decorators.py]: ah yes, did you correct that in the end?!

File: syncopy/tests/test_decorators.py [hunk 4]
```diff
@@ -156,82 +158,93 @@ def test_invalidcallstyles(self):
...
+
             assert groupList == fnameList
 
             # 1. cfg positional, 2. data positional
-            fnameList = group_objects(cfg, *self.dataObjs)
+            fnameList = [group_objects(cfg, data) for data in self.dataObjs]
```
[the remark this answers is not in the document]
Comment (tensionhead) [reply syncopy/tests/test_decorators.py:211]: yeah.. that was really speed hacking :/

**Pull request merged**